model-engine/model_engine_server/common/dtos/llms/batch_completion.py

```diff
@@ -0,0 +1,294 @@
+from enum import Enum
+from typing import Dict, List, Optional, Union
+
+from model_engine_server.common.dtos.llms.chat_completion import (
+    ChatCompletionV2Request,
+    ChatCompletionV2Response,
+)
+from model_engine_server.common.dtos.llms.completion import (
+    CompletionV2Request,
+    CompletionV2Response,
+)
+from model_engine_server.common.pydantic_types import BaseModel, Field
+from typing_extensions import TypeAlias
+
+
+# Common DTOs for batch completions
+class ToolConfig(BaseModel):
+    """
+    Configuration for tool use.
+    NOTE: this config is highly experimental and signature will change significantly in future iterations.
+    """
+
+    name: str
+    """
+    Name of the tool to use for the batch inference.
+    """
+    max_iterations: Optional[int] = 10
+    """
+    Maximum number of iterations to run the tool.
+    """
+    execution_timeout_seconds: Optional[int] = 60
+    """
+    Maximum runtime of the tool in seconds.
+    """
+    should_retry_on_error: Optional[bool] = True
+    """
+    Whether to retry the tool on error.
+    """
+
+
+class BatchCompletionsModelConfig(BaseModel):
+    model: str = Field(
+        description="ID of the model to use.",
+        examples=["mixtral-8x7b-instruct"],
+    )
+
+    checkpoint_path: Optional[str] = Field(
+        default=None, description="Path to the checkpoint to load the model from."
+    )
+    seed: Optional[int] = Field(default=None, description="Random seed for the model.")
+
+
+class BatchCompletionsRequestBase(BaseModel):
```

Comment:
strictly speaking, I think "ideally" dtos should only be imported in the api layer, and anything from use case -> service/gateway/repo should be in some domain/entities; having said that, I don't think we're super strict about that in this codebase, so think this is fine. (and it saves on some repeating anyways)